### ISSUE TITLE
How to use Red - 1st Startup Tutorial

### ISSUE THREAD

**Issue by Chovin** [Cog-Creators/Red-DiscordBot#422]
Would be nice to have an interactive tutorial start up in Discord about how to use Red after the configuration process.

This issue will be the central point of communication for the development of this tutorial (at least until a PR is put up). Please leave comments/ideas/questions below. A PR will be put up shortly where collaboration can be more easily done. Feel free to put one up and reference it here if I'm too slow.
## Tutorial
- only automatically starts the 1st time Red starts and connects to its first server
- can be stopped at any time via `!tutorial` ?
- can be started again later via `!turorial` ?

All commands would be disabled to begin with and the Red would slowly enable them as it goes through:
- makes sure the owner is correct
- how `!help` works
  - how subcommands work
  - what Cogs are
    - list the loaded/unloaded cogs (this should be a default command) 
- `!cog` and link to approved repo list ?
- `!load` ?
- what mods/admins are cause for some reason that's a confusing topic
- link to docs and Red server
- if audio not **able** to be loaded, list general steps to fix it and point to #support

**Comment by palmtree5** [comment]
Hmm, wondering how best to implement this. One question: would this be done via PM or in a channel in the first server it joins? Because if it's being done via PM, the owner better already be correct before starting the tutorial

**Comment by Chovin** [comment]
I was thinking start in the server, confirm the owner, then say:

> Red looks all set. Wanna do the rest of the tutorial via PM or here? (choose by typing either `pm` or `here`) You can stop/restart the tutorial at any time by typing `!tutorial`

**Comment by Redjumpman** [comment]
I like this line of thinking. We should map out what all it goes through for the user. I would be happy to collab on this. Would be cool if we could have some statistics uploaded to see how many people go through aaand use the tutrial and how many people fully complete it. Would help determine its effectiveness. I think this would eliminate a lot of questions, but I would caution that we should not use this tutorial as a scapegoat for every question in the support channel. It should referenced as an additional tool of understanding the bot like how we link portions of the wiki.

**Comment by Kowlin** [comment]
a FAQ and Knowledge Base would also be amazing, but I doubt anyone has time/wants to write those.

**Comment by skeith** [comment]
@Kowlin I could try that FAQ and KB

**Comment by Kowlin** [comment]
PR's are always allowed on both here and the docs GH

**Comment by GhostofGoes** [comment]
Where are the docs currently? Is it just the wiki, or are there more places? Might do a FAQ/Knowledgebase/Startup tutorial in my free time, as I've learned a fair amount in the past 8-9 months of maintaining a Red.

**Comment by palmtree5** [comment]
the docs are at http://twentysix26.github.io/Red-Docs/ (the git repository is https://github.com/twentysix26/Red-Docs/tree/gh-pages)

**Comment by Nerdietony** [comment]
You beat me too it ..

**Comment by Chovin** [comment]
to revive this a bit, I wrote a little mock tutorial.. perhaps have it run when the bot joins its first server

Thoughts on putting something like this in V3?
I'll put a PR up some time next month when my schedule frees up if nobody gets to it first.

>Hi @ owner_or_server_owner! You've invited your Red-DiscordBot to its first server! 
It looks like you've set your prefix to `doop`. 
That means whenever you see `[p]` in the help docs, you can replace it with `doop`.
You can find out more about this in the Getting Started docs: https://twentysix26.github.io/Red-Docs/red_getting_started/
You can jump right in and start looking at commands by using `doophelp`,
or you can type `dooptutorial` to start the tutorial about how to use and find commands.
>
>\> dooptutorial
>
>All my commands work by preceding them with your prefix.
The most important command you'll be using is `[p]help`. Since `doop` is your prefix, you'd type `doophelp` to execute it.
Give it a shot!
\<waits for user...>
>
>Awesome! The bot shoulda sent you a list of a bunch of commands broken up into categories we call Cogs.
You can use `[p]help` to learn about cogs or specific commands by putting them after `[p]help`. That's how you send input to commands. You'll be doing this a lot.
>
>Let's find out more about the `General` cog. To do that, we do `[p]help General`. Try it!
\<waits..>
>
>oops, it looks like you actually typed the `[p]`. remember, `[p]` stands for your prefix, so you should use `doophelp General` instead! Try again!
\<waits..>
>
>Awesome! I just described the General cog as well as listed all the commands under it. Let's look at the `[p]flip` command, that looks interesting. use `[p]help flip` to see what it does!
\<waits..>
>
>Notice on the 1st line `[]` is surrounding `user`. The `[]` means that the `user` input is optional. If you see `<>` surrounding the input, then that means that that input is required! Sometimes optional arguments will have an equal sign `=` followed by something. This means that that is the default value that'll be used if you don't give it anything.
Let's try using the `[p]flip` command without any inputs.
\<waits...>
>
>Nice. Flip them tables!
Just like `[p]flip` was under the General cog, sometimes commands have other commands under them. We call them **subcommands** and the commands that hold subcommands, **command groups**. `[p]set` is an example of a command group with a few subcommands, and just like how we did `[p]help General` to see the commands under General, we can use `[p]help set` to see the commands under `[p]set`. Try it out!
\<waits..>
>
>Awesome, it looks like you can change my prefix with `[p]set prefix` if you'd like. Use `[p]help set prefix` to learn how.
That's pretty much it for commands! You should be able to find your way through anything now. Don't forget to go through the Getting Started guide if you get stuck https://twentysix26.github.io/Red-Docs/red_getting_started/
>
>Continue onto the 3rd Party Cog tutorial (y/n)?
...

**Comment by Redjumpman** [comment]
What is the status of this?

**Comment by Tobotimus** [comment]
https://twentysix26.github.io/Red-Docs/red_getting_started/ Looks like it's all good to me. @retke has also included a getting started guide in #1757. So I'm gonna go ahead and close this :)
